Rustc incorrectly points to `where`-clause bound when calling static method on invalid type

The following code:

```rust
trait Foo {
    fn my_method() where Self: Clone;
}

fn main() {
    <bool as Foo>::my_method();
}
```

produces the following output:

```
error[E0277]: the trait bound `bool: Foo` is not satisfied
 --> src/main.rs:6:5
  |
2 |     fn my_method() where Self: Clone;
  |                                ----- required by this bound in `Foo::my_method`
...
6 |     <bool as Foo>::my_method();
  |     ^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Foo` is not implemented for `bool`

error: aborting due to previous error
```

The root cause of the error is the fact that `bool` does not implement `Foo`, so the `Self: Clone` bound isn't relevant. If the `where` clause is split over multiple lines, then this can obscure the actual cause of the error.

--- a/tests/ui/trait-bounds/issue-82038.rs
+++ b/tests/ui/trait-bounds/issue-82038.rs
@@ -1,0 +1,9 @@
+// Failed bound `bool: Foo` must not point at the `Self: Clone` line
+
+trait Foo {
+    fn my_method() where Self: Clone;
+}
+
+fn main() {
+    <bool as Foo>::my_method(); //~ERROR [E0277]
+}

--- a/tests/ui/trait-bounds/issue-82038.stderr
+++ b/tests/ui/trait-bounds/issue-82038.stderr
@@ -1,0 +1,9 @@
+error[E0277]: the trait bound `bool: Foo` is not satisfied
+  --> $DIR/issue-82038.rs:8:6
+   |
+LL |     <bool as Foo>::my_method();
+   |      ^^^^ the trait `Foo` is not implemented for `bool`
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0277`.

